src/backend/mod.rs

```diff
@@ -66,6 +70,39 @@ pub fn init_backend_auto(
             .seats
             .add_seat(initial_seat);
 
+        let desired_numlock = state
+            .common
+            .config
+            .cosmic_conf
+            .keyboard_config
+            .numlock_state;
+        // Restore numlock state based on config.
+        let toggle_numlock = match desired_numlock {
+            NumlockState::BootOff => keyboard.modifier_state().num_lock,
+            NumlockState::BootOn => !keyboard.modifier_state().num_lock,
+            NumlockState::LastBoot => {
+                keyboard.modifier_state().num_lock
+                    != state.common.config.dynamic_conf.numlock().last_state
+            }
+        };
+
+        // If we're enabling numlock...
+        if toggle_numlock {
+            let mut input = |key_state| {
+                let time = state.common.clock.now().as_millis();
+                let _ = keyboard.input(
+                    state,
+                    smithay_input::Keycode::new(77),
```

Comment:
I really dislike magic numbers, can you please add a `const KEY_NUMLOCK = 69` to the beginning of this file and do `KEY_NUMLOCK + 8` here?

Reply:
For me, `KEY_NUMLOCK + 8` is equally magical as  `77` is